### PR TITLE
Update DLint

### PR DESCRIPTION
**Pull request by ghost**
Update to latest DLint. Adds pattern rules. Tests coming in next PR.

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/Generics.daml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 -- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
+-- @INFO Redundant case
```
Comment (cocreature) [inline compiler/damlc/tests/daml-test-files/Generics.daml]: Where is the redundant case here? I’m having trouble seeing any case statement.

File: compiler/damlc/tests/daml-test-files/LogicTest.daml [hunk 2]
```diff
@@ -14,7 +14,10 @@ propositionTest = scenario do
     r = Proposition 3
     f : Formula Int = false
     t : Formula Int = true
-    truth i = if i == 1 then Some True else if i == 2 then Some False else None
+    truth i
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply compiler/damlc/tests/daml-test-files/LogicTest.daml]: No, that's a new DLint hint!

[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply compiler/damlc/tests/daml-test-files/LogicTest.daml]: Checked in with Bernhard, he likes it too 😄

File: compiler/damlc/tests/daml-test-files/Generics.daml (hunk 1 above)
Comment (shayne-fletcher-da) [reply compiler/damlc/tests/daml-test-files/Generics.daml]: As we have determined, this is in generated code. Fixing upstream (testing now). Will withdraw this PR most likely. Stand by.

**Comment by shayne-fletcher-da** [comment]
Sent a patch upstream to HLint that solves the generics issue. Withdrawing this PR until that lands.

**Pull request closed without merging**